src/set_theory/ordinal/natural_ops.lean

```diff
@@ -0,0 +1,354 @@
+    { exact lt_max_of_lt_right ((hf hi (add_lt_add_left hb' a) hi').trans_lt (lt_blsub _ b' _)) } },
+  all_goals
+  { apply blsub_le_of_brange_subset.{u u v},
+    rintro c ⟨d, hd, rfl⟩,
+    apply mem_brange_self }
+end
+
+private theorem add_assoc' : ∀ a b c : nat_ordinal, a + b + c = a + (b + c)
+| a b c := begin
+  nth_rewrite 2 add_def,
+  rw [add_def, blsub_add_of_mono, blsub_add_of_mono, max_assoc],
+  { congr;
+    ext d hd;
+    apply add_assoc' },
+  { exact λ i j _ _ h, add_le_add_left h a },
+  { exact λ i j _ _ h, add_le_add_right h c }
+end
+using_well_founded { dec_tac := `[solve_by_elim [psigma.lex.left, psigma.lex.right]] }
+
+instance : add_semigroup nat_ordinal :=
+{ add := (+),
+  add_assoc := add_assoc' }
+
+private theorem add_comm' : ∀ a b : nat_ordinal, a + b = b + a
+| a b := begin
+  rw [add_def, add_def, max_comm],
+  congr;
+  ext a' ha';
+  apply add_comm'
+end
+using_well_founded { dec_tac := `[solve_by_elim [psigma.lex.left, psigma.lex.right]] }
+
+instance : add_comm_semigroup nat_ordinal :=
+{ add_comm := add_comm',
+  ..nat_ordinal.add_semigroup }
+
+private theorem add_zero' (a : nat_ordinal) : a + 0 = a :=
+begin
+  induction a using nat_ordinal.induction with a IH,
+  rw add_def,
+  simp only [blsub, of_nat_ordinal_zero, blsub_zero, to_nat_ordinal_zero, max_zero_right],
+  rw ←eq_of_nat_ordinal_iff,
+  convert blsub_id (of_nat_ordinal a),
+  ext b hb,
+  exact IH _ hb
+end
+
+instance : add_cancel_comm_monoid nat_ordinal :=
+{ zero := 0,
+  add_zero := add_zero',
+  zero_add := λ a, by rw [add_comm, add_zero'],
+  add_left_cancel := λ a b c, add_left_cancel'',
+  ..nat_ordinal.add_comm_semigroup }
+
+/-- The successor on `nat_ordinal` is still `a + 1`. -/
+theorem add_one (a : nat_ordinal) : a + 1 = to_nat_ordinal (succ (of_nat_ordinal a)) :=
+begin
+  induction a using nat_ordinal.induction with a IH,
+  rw add_def,
+  simp only [blsub, of_nat_ordinal_one, blsub_one, to_nat_ordinal_zero, add_zero, max_eq_right_iff,
+    to_nat_ordinal_le_iff, blsub_le_iff],
+  intros i hi,
+  rw IH (to_nat_ordinal i) hi,
+  exact succ_lt_succ.2 hi
+end
+
+theorem one_add (a : nat_ordinal) : 1 + a = to_nat_ordinal (succ (of_nat_ordinal a)) :=
+by rw [add_comm, add_one]
+
+theorem add_nat (a : nat_ordinal) (n : ℕ) : a + n = to_nat_ordinal (of_nat_ordinal a + n) :=
+begin
+  induction n with n hn,
+  { simp },
+  { rw [nat.cast_succ, nat.cast_succ, ←add_assoc, ←add_assoc, hn, add_one],
+    refl }
+end
+
+theorem nat_add (n : ℕ) (a : nat_ordinal) : ↑n + a = to_nat_ordinal (of_nat_ordinal a + n) :=
+by rw [add_comm, add_nat]
+
+@[simp] theorem of_nat_ordinal_cast (n : ℕ) : of_nat_ordinal (↑n) = ↑n :=
+begin
+  induction n with n hn,
+  { refl },
+  { simpa [add_one, hn] }
+end
+
+@[simp] theorem to_nat_ordinal_cast (n : ℕ) : to_nat_ordinal (↑n) = ↑n :=
+by { rw ←of_nat_ordinal_cast n, refl }
+
+theorem ordinal_add_le_add (a b : ordinal.{u}) :
+  a + b ≤ of_nat_ordinal (to_nat_ordinal a + to_nat_ordinal b) :=
```

Comment:
```suggestion
theorem to_nat_ordinal_add_le (a b : ordinal.{u}) :
  to_nat_ordinal (a + b) ≤ to_nat_ordinal a + to_nat_ordinal b) :=
```
or
```suggestion
theorem le_of_nat_ordinal_add (a b : nat_ordinal.{u}) :
  of_nat_ordinal a + of_nat_ordinal b ≤ of_nat_ordinal (a + b) :=
```
(you should have both, up to you which is easier to prove first)

Reply:
I ended up writing this as just `a + b ≤ a ♯ b`.